### PR TITLE
compaction: add formatter for map<timestamp_type, vector<shared_sstab…

### PR DESCRIPTION
**Pull request by tchaikov**
…le>>

before this change, we rely on the default-generated fmt::formatter created from operator<<, but fmt v10 dropped the default-generated formatter.

in this change, we define a formatter for
map<timestamp_type, vector<shared_sstable>>. since the operator<< for this type is only used in the .cc file, and the only use case of it is to provide the formatter for fmt, so the operator<< based formatter is remove in this change.

Refs #13245
Signed-off-by: Kefu Chai <kefu.chai@scylladb.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> 
> ✅ - Build ❌ - Sanity Tests ✅ - Unit Tests
> #### Failed Tests (1/21866):
> 
>     * [test_numeric_type_group](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4871/testReport/junit/cql_group_functions_tests/TestGroupFunctions/Tests___Sanity_Tests___test_numeric_type_group) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_numeric_type_group)
> 
> 
> #### Build Details:
> 
>     * Build URL: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/4871/
> 
>     * Duration: 2 hr 19 min
> 
>     * Builder: spider2.cloudius-systems.com

new flaky test: #16168

File: compaction/time_window_compaction_strategy.cc [hunk 1]
```diff
@@ -433,15 +433,22 @@ time_window_compaction_strategy::get_buckets(std::vector<shared_sstable> files,
...
-    os << "  buckets = {\n";
-    for (auto& bucket : buckets | boost::adaptors::reversed) {
-        os << format("    key={}, size={}\n", bucket.first, bucket.second.size());
-    }
-    os << "  }\n";
-    return os;
 }
 
+template <>
+struct fmt::formatter<std::map<sstables::timestamp_type, std::vector<sstables::shared_sstable>>> {
```
Comment (denesb) [inline compaction/time_window_compaction_strategy.cc:439]: Doesn't format provide formatters for containers?

Comment (tchaikov) [reply compaction/time_window_compaction_strategy.cc:439]: actually, we have two set of formatter formatting `std::map`. one from `utils/to_string.hh`, the other from `fmt/ranges.h`. but in this case, we customize the formatter by adding, for instance, `buckets = ` at the beginning of the output, and print the values like `key={key}, size={value.size)`. this is something neither of the homebrew generic formatter or the fmt/ranges.hh can offer.

hence this specialization.

Comment (denesb) [reply compaction/time_window_compaction_strategy.cc:439]: I see, fair enough.

**Pull request closed without merging**